webcompat/static/js/lib/issue-list.js

```diff
@@ -191,42 +188,53 @@ issueList.SearchView = Backbone.View.extend({
   updateSearchQuery: function(data) {
     this.input.val(data);
   },
-  _isEmpty: true,
-  _currentSearch: null,
-  searchIfNotEmpty: _.debounce(function(e) {
+  doSearch: _.debounce(function(e) {
     var searchValue = e.type === 'click' ? $(e.target).prev().val() :
                       e.target.value;
-    if (searchValue.length) {
-      this._isEmpty = false;
-      // don't search if nothing has changed
-      // (or user just added whitespace)
-      if ($.trim(searchValue) !== this._currentSearch) {
-        this._currentSearch = $.trim(searchValue);
-        this.doSearch(this._currentSearch);
-        // clear any filters that have been set.
-        issueList.events.trigger('filter:clear', {removeQ: false});
-      }
-    }
+    this.mainView.params.setParam('search', searchValue);
+  }, 250),
+});
 
-    // if it's empty and the user searches, show the default results
-    // (but only once)
-    if (!searchValue.length && this._currentSearch !== '') {
-      this._currentSearch = '';
-      this._isEmpty = true;
-      issueList.events.trigger('issues:update');
-    }
-  }, 350),
-  doSearch: _.throttle(function(value) {
-    if (!this._isEmpty) {
-      issueList.events.trigger('issues:update', {query: value});
-    }
-  }, 500)
+/*
+* The applied labels view shows the labels we're filtering by - *excluding*
+* status- prefixed labels. These labels can be un-set by clicking their X
+* or by focusing them by keyboard and pressing del
+*/
+issueList.AppliedLabelsView = Backbone.View.extend({
+  mainView: null,
+  el: $('.wc-AppliedLabels .wc-content'),
+  events: {
+    'click .wc-Filter--appliedlabel': 'deleteLabel'
+  },
+  keyboardEvents: {
+    'delete': 'deleteLabel'
+  },
+  initialize: function(options) {
+    this.mainView = options.mainView;
+    issueList.events.on('appliedlabels:update', _.bind(this.updateAppliedLabels, this));
+
+  },
+  template: _.template($('#issuelist-appliedlabels-tmpl').html()),
+  updateAppliedLabels: function() {
+    var labels = this.mainView.params.get('label');
+    this.render({labels:labels});
+  },
+  render: function(options) {
+    this.$el.html(this.template(options));
+    return this;
+  },
+  deleteLabel: function(e) {
+    var label = e.target.firstChild.textContent.trim();
```

Comment:
If we're going to support old browsers I'd prefer adding String.prototype.trim=.. somewhere :)